src/java/org/apache/cassandra/schema/Types.java

```diff
@@ -506,6 +587,68 @@ public long serializedSize(Types t, Version version)
                 size += sizeof(fieldTypes.size());
                 for (String s : fieldTypes)
                     size += sizeof(s);
+                if (version.isAtLeast(Version.V8))
+                {
+                    size += BOOL_SIZE;
+                    if (!StringUtils.isEmpty(type.comment))
+                        size += sizeof(type.comment);
+                    size += BOOL_SIZE;
+                    if (!StringUtils.isEmpty(type.securityLabel))
+                        size += sizeof(type.securityLabel);
+
+                    size += fieldMetadataSize(type, fieldNames);
+                }
+            }
+            return size;
+        }
+
+        private void serializeFieldMetadata(UserType type, List<String> fieldNames, DataOutputPlus out) throws IOException
+        {
+            out.writeInt(fieldNames.size());
+            for (String fieldName : fieldNames)
+            {
+                FieldIdentifier fieldId = FieldIdentifier.forUnquoted(fieldName);
```

Comment:
you are calling `StringUtils.isEmpty(fieldSecurityLabel)` four times in this method, the minimum number of calls is 2.